enhancing/enhancer.js

```diff
@@ -2,21 +2,36 @@ module.exports = {
   succeed,
   fail,
   repair,
-  get,
+  get
 };
 
 function succeed(item) {
-  return { ...item };
+  return {
+    ...item,
+    enhancement: item.enhancement < 20 ? item.enhancement + 1 : item.enhancement
+  };
 }
 
 function fail(item) {
-  return { ...item };
+  return {
+    ...item,
+    durability:
+      item.enhancement < 15 ? item.durability - 5 : item.durability - 10,
```

Comment:
here's an interesting edge case you might want to account for--what if this decrementing drops the item's durability below zero?